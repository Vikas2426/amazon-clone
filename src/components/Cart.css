.cart-item {
  display: flex;
  align-items: center;
}

.cart-item-img {
  width: 100px;
}
