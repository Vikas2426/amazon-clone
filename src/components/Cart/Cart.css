.cart{
  background-color: white;
}
.cart-banner{
  width: 100vw;
  max-height:20vh;
}
.cart-heading{
  border-bottom: 2px solid grey;
  padding: 2%;
  display: flex;
  justify-content: space-between;
}
.empty-cart-icon{
  text-align: center;
  font-size: 10em;
  color: rgba(107, 107, 207, 0.308);
}
.cart-item {
  display: flex;
  align-items: center;
  width: 100vw;
  background-color: white;
  padding:2%;
  margin-bottom: 0.5%;
  border-bottom: 2px solid rgb(182, 182, 182);
}

.cart-item-details{
  min-height: 150px;
  margin-left: 0.5%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-self: flex-start;

}
.cart-item-img {
  max-width: 200px;
  margin-right: 5%;
}

.change-quantity-btn{
background-color: rgb(209, 209, 209);
font-size: medium;
color: black;
width: 25px;
height: 25px;
}

.remove-item-btn{
    background-color: yellow;
    color: black;
    font-size: small;
    padding: 10px 20px;
    align-self: center;
    font-weight: 500;
}
