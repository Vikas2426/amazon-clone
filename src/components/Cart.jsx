import React, { useContext } from 'react';
import { itemsContext } from './ContextProvider';
import Header from './Header';
import './Cart.css';

const Cart = () => {
    const items = useContext(itemsContext);
    return <div className="Cart">
        <Header />

        {items.length ? items.map(item => (
            <div className='cart-item'>
                <img src={item.imgUrl} alt='item' className='cart-item-img' />
                <div className='cart-item-details'>
                    <p className='cart-item-price'>${item.price}</p>
                    <p>Quantity: {item.quantity}</p>
                </div>
                {/* <button disabled>Remove</button> */}
            </div>
        ))

            : <h2 style={{ textAlign: 'center' }}>No item added to cart yet!</h2>
        }
    </div>
}

export default Cart;