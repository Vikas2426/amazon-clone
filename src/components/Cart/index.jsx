import React, { useContext } from "react";
import CartItem from "../CartItem/CartItem";
import { cartContext } from "../ContextProvider";
import Header from "../Header";
import Total from "../Total";
import "./Cart.css";
import { useTransition, animated } from "react-spring";
import banner from "../../assets/cart-banner.jpg";

const Cart = () => {
  const { items } = useContext(cartContext);

  const slideRight = useTransition(items, {
    enter: { x: "0%", postiion: "relative" },
    leave: { x: "100vw", position: "relative" },
    config: { duration: 400 },
  });

  return (
    <div className="Cart">
      <Header />
      <img className="cart-banner" src={banner} alt="cart-banner" />
      {items.length ? (
        <h2 className="cart-heading">
          Your Cart: <Total />
        </h2>
      ) : (
        <h2 className="cart-heading">Your Cart: is empty</h2>
      )}
      {!items.length ? (
        <h1 className="empty-cart-icon">
          <i class="fas fa-shopping-cart"></i>
        </h1>
      ) : (
        slideRight((styles, item) => (
          <animated.div className="cart-item" key={item.imgUrl} style={styles}>
            <CartItem item={item} />
          </animated.div>
        ))
      )}
    </div>
  );
};

export default Cart;
