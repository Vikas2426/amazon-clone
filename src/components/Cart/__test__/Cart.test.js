import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Cart from "..";

jest.mock("../../Header", () => "header");

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
    items: [
      {
        imgUrl: "www.url.com",
        price: 400,
        quantity: 1,
      },
    ],
  }),
}));

describe("Cart", () => {
  it("should render without error", () => {
    render(<Cart />);
    expect(screen).toMatchSnapshot();
  });
});
