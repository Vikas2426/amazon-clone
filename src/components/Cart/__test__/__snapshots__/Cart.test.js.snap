// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should render without error 1`] = `Screen {}`;
